rsc-server/src/org/openrsc/model/NpcManager.java

```diff
@@ -0,0 +1,148 @@
+package org.openrsc.model;
+
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.logging.Logger;
+
+/**
+ * Manages the npc instances.
+ */
+public class NpcManager {
+
+	private static final NpcManager INSTANCE = new NpcManager();

```

Comment:
I prefer the singleton design, most of my projects are like this and I've never had issues; as long as instances weren't passed as parameters before being constructed.
I'd rather keep it as is, unless a problem occurs. If there is an issue later on I'll rewrite it, but I think it will be fine in this project.